java-checks/src/test/files/checks/DanglingElseStatementsCheck.java

```diff
@@ -0,0 +1,34 @@
+class A {
+  void f() {
+    boolean a = true, b = false;
+    int d = 0, e = 0;
+
+    if (a)
+      d++;
+
+    if (a)
+      d++;
+    else        // Compliant, not nested
+      e++;
+
+    if (a)
+      d++;
+    else if (b) // Compliant, else if construct
+      e++;
+    else
+      d = e;
+
+    if (a)
+      if (b) {
+        d++;
+      } else {  // Compliant, curly braces
+        e++;
+      }
+
+    if (a)
+      if (b)
+        d++;
+    else        // Noncompliant {{Add explicit curly braces to avoid dangling else.}}
+      e++;
+  }
+}
```

Comment:
You should configure your editor to insert empty line at the end of file, see here https://stackoverflow.com/questions/44704968/visual-studio-code-insert-new-line-at-the-end-of-files